builtin/providers/aws/resource_aws_key_pair.go

```diff
@@ -27,10 +27,24 @@ func resourceAwsKeyPair() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"key_name": &schema.Schema{
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"key_name_prefix"},
+			},
+			"key_name_prefix": &schema.Schema{
 				Type:     schema.TypeString,
 				Optional: true,
-				Computed: true,
 				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) > 100 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 100 characters, name is limited to 255", k))
```

Comment:
This matches the validation for the `name_prefix` argument for `aws_security_group` resource. I have double-checked and the maximum key pair name is 255 characters.

<img width="539" alt="screen shot 2016-11-10 at 12 30 06 am" src="https://cloud.githubusercontent.com/assets/121107/20139947/a3b7f5fc-a6dc-11e6-88bf-4ed9c0b067bb.png">